wsynphot/get_filter_data.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import pandas as pd
+import requests, tempfile
+
+from astropy.io.votable import parse_single_table
+# The VOTables fetched from SVO contain only single table element, thus parse_single_table
+
+svo_main_url = 'http://svo2.cab.inta-csic.es/theory/fps/fps.php'
+
+# A utility fuction to convert byte strings to normal literal strings
+def byte_to_literal_strings(dataframe):
+    # Select the str columns:
+    str_df = dataframe.select_dtypes([np.object])
+    
+    # Convert all of them into unicode strings
+    str_df = str_df.stack().str.decode('utf-8').unstack()
+    
+    # Swap out converted cols with the original df cols
+    for col in str_df:
+        dataframe[col] = str_df[col]
+    return dataframe
+
+
+def get_filter_index():
+    # To get master list, fetch all filters from SVO of wavelength between 0 & very large no.
+    query = {'WavelengthEff_min': 0,
+             'WavelengthEff_max': 10000000}
+    response = requests.get(svo_main_url, params=query)
+    response.raise_for_status()
+    
+    # Save the fetched VOTable (XML file) as a temporary file and parse it into an Astropy table
+    with tempfile.NamedTemporaryFile() as votable:
```

Comment:
we can directly load this - without writing to disk

Reply:
No actually I've tried this. This VOTable parsing function only takes a file path as input - otherwise gives error. So I had to save it - but I've made sure it won't be visible to user because tempfile gets deleted automatically as soon as they are closed.

Reply:
OK used stream: `io.BytesIO` instead!